api/lib/infrastructure/serializers/jsonapi/campaign-serializer.js

```diff
@@ -5,7 +5,7 @@ const Campaign = require('../../../domain/models/Campaign');
 
 module.exports = {
 
-  serialize(campaigns, { tokenForCampaignResults, ignoreCampaignReportRelationshipData = true } = {}) {
+  serialize(campaigns, meta, { tokenForCampaignResults, ignoreCampaignReportRelationshipData = true } = {}) {
```

Comment:
J'ai pas l'habitude de cette syntaxe. Elle est peu répandue, non ? Est ce qu'elle a un intérêt par rapport à `serialize: () => {}`. Peut être une histoire de scope lié à `jsonapi` ?

Reply:
C'est du sucre. Pourquoi se faire mal à mettre les `: `et la `=> ` alors qu'on peut s'en passer ?

Reply:
Je ne connais pas la différence. Je ne suis pas sûre que ce soit que du sucre. D'instinct je dirais que dans l'une le this est pas bindé ? Mais en a-t-on besoin ? C'est intéressant à creuser :)

Reply:
@lisequesnel `{ serialize() { … } }` est équivalent à `{ serialize: function() { … } }`. Effectivement c'est différent de `{ serialize: () => { … } }` au niveau du _binding_ de `this`. Du point de vue de l'appelant, je trouve que la version qui n'utilise pas `=>` sera moins surprenante.

Reply:
En fait les arrow functions `() => {}` et les functions `function name() {}` ont chacune des spécificités que ce soit en terme de binding du this ou de fonctionnalités supplémentaires comme le hoisting avec le mot clé `function`. Est ce que cette troisième syntaxe a des avantages ou particularités ?

Reply:
@frinyvonnick il n'y a pas de _hoisting_ pour les _function expressions_ 😉 

L'intérêt de la "nouvelle" (ECMAScript 2015) syntaxe `{ serialize() { } }` me paraissait évident : c'est plus court ? Quant à ses particularités, ce sont exactement celles de son équivalent ([tel que généré par Babel par exemple](https://babeljs.io/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBAhjAvDA3gKBjAHgCgJSowBOAplAK5FgxQAWAlhAHQIC-ANGq0A&debug=false&forceAllTransforms=true&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=false&presets=env%2Cenv&prettier=false&targets=&version=7.8.4&externalPlugins=)) `{ serialize: function serialize() { } }` donc rien de nouveau sous le soleil.

Reply:
On a écrit nos commentaires en même temps, je n'avais pas vu ta réponse à @lisequesnel @jonathanperret.